ODIN_II/verify_odin.sh

```diff
@@ -881,12 +881,20 @@ function sim() {
 
 				run_this_test="on"
 
-				if [ "_${_SUBTEST_LIST[*]}" != "_" ] \
-				&& ! echo "${_SUBTEST_LIST[@]}" | grep "${TEST_FULL_REF}" &> /dev/null;
+				if [ "_${_SUBTEST_LIST[*]}" != "_" ]
 				then
-					# skip duplicate tests
 					run_this_test="off"
-				elif [ -d "${NEW_RUN_DIR}/${TEST_FULL_REF}" ]
+					for subtest in "${_SUBTEST_LIST[@]}";
+					do
+						if echo "_${TEST_FULL_REF}" | grep -E "_${subtest}" &> /dev/null
```

Comment:
if you dropped the underscore, i think you could just grep the whole set, you wouldnt have to specify the suite_name either.

Reply:
Im not sure what you mean? The `_SUBTEST_LIST[@]` is the list of regular expressions so I cant grep that set. The underscore seems unnecessary I agree, but I am not sure how that allows us to avoid needing the suite_name

Reply:
I believe that if you also get rid of some of this, it will get rid of the requirement of the suite being passed in. 
https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/3e17e6dfa1cf79e9b925bf764b2880f51bbae2fb/ODIN_II/verify_odin.sh#L695-L715
you will need to check that at least one test was run for the task before continuing on with the parsing phase, but this would let you use just a regex without specifying the task name.

Reply:
```bash
 ########################################## 
 # check if we only run some subtask 
 run_benchmark="on" 
 run_subtest_only="" 
 if [ "_${_SUBTEST_LIST[*]}" != "_" ]; 
 then 
 	run_subtest_only="--subset" 
 fi 
```

Reply:
Yeah already done

Reply:
does it work? (you can pass --dry_run to not wait through all the shenanigans)

Reply:
Looks like
`./verify_odin.sh -t regression_test/benchmark/task/syntax *bm_* *8_bit*`
Runs only the tests matching those regex

Reply:
And
`./verify_odin.sh -t regression_test/benchmark/task/syntax *some_unmatchable_regex*`
Runs nothing and exits without error

Reply:
> does it work? (you can pass --dry_run to not wait through all the shenanigans)

tested locally and it does :)

Reply:
>Runs nothing and exits without error

looks perfect to me then! 
thanks for this! a much needed upgrade!